### ISSUE TITLE
Search functionality on Apps Managements screen is crazy slow/laggy

### ISSUE THREAD

**Issue by carlylemiii** [AdguardTeam/AdguardForAndroid#2951]
### Steps to reproduce
1. Open Apps Management
2. Tap `Search app…` button
3. Notice Search is slow to load.
4. Tap the name of an app you have installed.
5. Watch as what you've typed at a normal speed finally catches up and is displayed in AdGuard.
6. Tap on the result.
7. Result is slow to load, and before loading, you're shown the full list of apps again in Apps Management.
8. Search for another app.
9. Tap back arrow and notice that backwards navigation is even slow.
10. Tapping an app from the full list is even a little slow - not painfully so, not like in the above scenarios, but enough that it's a noticeable delay.

### Expected behavior
Speed!

### Actual behavior
_Sooo_ slow it's like I'm running the app on a device from five years ago.

### Your environment

* Adguard version: 3.2.98β
* Adguard filtering mode: Local VPN
* Device model: Pixel 3 XL
* Operating system and version: Android 9
* Root access: No
* Any other apps that can start VPN / Antiviruses / Cleaners etc. installed?: No

**Comment by zzebrum** [comment]
I agree with step 5. Other steps look not that bad on my device though. 
@TheHasagi please check on other devices.

**Comment by carlylemiii** [comment]
Let me know if you need video of any of the others. 2-3 should be painfully obvious, too, the rest are less drastic but still quite reproduceable.

**Comment by ameshkov** [comment]
Actually, it is really very slow.

@artemiv4nov please check what's done when the search is performed.

I am starting to think that we are re-executing SQL queries for the apps' stats every time the text is changed in the search input. This may be fast for newly installed apps, but when you have a lot of stats to select, this will be super slow.

**Comment by carlylemiii** [comment]
I saw this mentioned as an improvement in the release notes for v3.2.117β but I cannot tell a difference at all for what it's worth.

**Comment by ameshkov** [comment]
@carlylemiii could you please try the nightly build? I am on .119 and I don't see any issues.

**Comment by carlylemiii** [comment]
@ameshkov it is still very slow in the nightly. Note how long it takes for the search to open after I press the search button (that's not me long-pressing; that's just it taking a while to respond). Then, note how much of a delay there is between my keyboard input and that being displayed in the search bar (including backspacing, as well). Point number 7 above still is true, too, though it's not captured in this recording I just made, as I skipped that part.

https://www.dropbox.com/s/w3ghdsh9cz7mop9/20190805_002757.mp4?dl=0
